### DISCUSSION TITLE
Sonos Timer

### DISCUSSION THREAD

**Discussion by lou-cucci** [punxaphil/custom-sonos-card#78] [Q&A] [answered]
Love the custom-sonos-card but would like to add the sonos sleep timer to the controls... I can see it on one of your screenshots but can't find the code...how can I do that please?

**Comment by punxaphil** [comment]
Which screenshot?

**Comment by lou-cucci** [comment]
https://raw.githubusercontent.com/johanfrick/custom-sonos-card/master/img/dark-favorites.png

**Reply by lou-cucci** [reply]
The Z I was assuming is a sleep timer

**Reply by punxaphil** [reply]
I'm afraid not, that is just a Sonos feature to enable night sound for the Beam speaker.

**Comment by lou-cucci** [comment]
Thanks for looking into this Johan

Appreciate all the work you do for HA

**Comment by jarvih** [comment]
Hi,

I whould like to have sonos sleep timer controls on card.

**Reply by punxaphil** [reply]
Home assistant does not support it: https://www.home-assistant.io/integrations/sonos/. So I can't add it to the card.

**Comment by jarvih** [comment]
Sure it does.

"SERVICE SONOS.SET_SLEEP_TIMER" and "SERVICE SONOS.CLEAR_SLEEP_TIMER"

I have used it on automations but it would be nice to have manual control on card also.

**Reply by punxaphil** [reply]
My mistake!
Not sure how such a gui would look though

**Reply by jzastoupil** [reply]
Something like this?
https://community.home-assistant.io/t/sonos-sleep-timer-template/409383/7

**Reply by punxaphil** [reply, marked as answer]
Already in the card

https://github.com/punxaphil/custom-sonos-card/issues/540

**Reply by jzastoupil** [reply]
Thanks for the link! I saw mention of a sleep timer in the readme but couldn't find any documentation on how to implement it. I didn't realize it was buried in the volume settings. I was hoping that I could put it in a card on my screen that was basically just the sleep timer so I could one-touch set it when I'm going to bed instead of having to click multiple times. (Maybe I can do this? I just started using it so I'm playing around)

Still I am grateful it exists!
